Compile UserSpaceInstrumentation lib with hidden symbols (#3857)
google/orbit: src/UserSpaceInstrumentation/CMakeLists.txt

# executed on function entry/exit of an instrumented function.
add_library(OrbitUserSpaceInstrumentation SHARED)

# The linker flags mark all symbols coming from libraries as private and ensure that they will be hidden in the shared library.
# The visibility properties ensure that symbols coming from this module's source files are considered private unless

> # explicitly marked visible in the source code file.

Reviewer: In my opinion what is missing in this comment is _why_ we need the symbols to be hidden/private (it's explained in the commit message but I think it should be explain here as well).

Author: Good point. Done. Let me know what you think.

# executed on function entry/exit of an instrumented function.
add_library(OrbitUserSpaceInstrumentation SHARED)

# The linker flags mark all symbols coming from libraries as private and ensure that they will be hidden in the shared library.
# The visibility properties ensure that symbols coming from this module's source files are considered private unless
# explicitly marked visible in the source code file.

> # This avoids functions and types that we don't want to explicitly expose leaking into the target process and avoids
> # them potentially being used outside of this library.

Reviewer: I would suggest the following, to separate "expose" and "leaking", which belong to different sentences, and because I think "avoid"+"-ing" is used when the second verb also refers to the subject of "avoid":
```suggestion
# This prevents functions and types that we don't want to explicitly expose from leaking into the target process and from
# potentially being used outside of this library.
```

Author: Update Text. Thanks, Davide!
